compiler/rustc_resolve/src/imports.rs

```diff
@@ -828,16 +828,23 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                     Err(Undetermined) => indeterminate_count += 1,
                     // Don't update the resolution, because it was never added.
                     Err(Determined) if target.name == kw::Underscore => {}
-                    Ok(binding) if binding.is_importable() => {
+                    Ok(binding)
+                        if binding.is_importable()
+                            || this.tcx.features().import_trait_associated_functions() =>
```

Comment:
The RFC says only associated functions and constants can be imported, not however associated types. You should exclude associated types.

<details><summary>(In debug builds, they lead to an ICE in your current implementation)</summary>

```rs
trait Trait { type Assoc; }
use Trait::Assoc;
type Alias = Assoc;
```

</details>